core/src/main/java/org/apache/calcite/rel/rules/LoptOptimizeJoinRule.java

```diff
@@ -2071,12 +2071,12 @@ private static boolean areSelfJoinKeysUnique(RelMetadataQuery mq,
     for (IntPair pair : joinInfo.pairs()) {
       final RelColumnOrigin leftOrigin =
           mq.getColumnOrigin(leftRel, pair.source);
-      if (leftOrigin == null || !leftOrigin.isDerived()) {
```

Comment:
Could you give some code comments explaining why you changed this code?

Code comments could help people better to understand why you made a change.

Reply:
Update the code, thanks